-Wmaybe-uninitialized warning for SolverComparer

<!--Please only submit issues/bug reports that come with enough information to reproduce them, ideally a unit test that fails, and possible ideas on what might be wrong. -->

<!-- Even better yet, fix the bug and/or documentation, add a unit test, and create a pull request! -->

<!-- This is a channel to report bugs/issues, not a support channel to help install/use/debug your own code. We'd love to help, but just don't have the bandwidth. Please post questions in the GTSAM Google group (https://groups.google.com/forum/#!forum/gtsam-users) -->

## Description

<!-- A clear description of the bug -->
-Wmaybe-unitialized warning from a Key variable.

## Steps to reproduce

1. git clone https://github.com/borglab/gtsam
2. mkdir build && cd build && cmake .. && make

<!-- If you have a code sample, unit test, error messages, stack traces, etc., please provide it here as well -->

```
In file included from /home/acxz/vcs/git/github/acxz/gtsam/gtsam/nonlinear/Values.h:544,
                 from /home/acxz/vcs/git/github/acxz/gtsam/gtsam/nonlinear/NonlinearFactor.h:23,
                 from /home/acxz/vcs/git/github/acxz/gtsam/gtsam/slam/BetweenFactor.h:22,
                 from /home/acxz/vcs/git/github/acxz/gtsam/examples/SolverComparer.cpp:34:
/home/acxz/vcs/git/github/acxz/gtsam/gtsam/nonlinear/Values-inl.h: In function ‘void runIncremental()’:
/home/acxz/vcs/git/github/acxz/gtsam/gtsam/nonlinear/Values-inl.h:382:11: warning: ‘firstPose’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  382 |     insert(j, static_cast<const Value&>(GenericValue<ValueType>(val)));
      |     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/acxz/vcs/git/github/acxz/gtsam/examples/SolverComparer.cpp:258:7: note: ‘firstPose’ was declared here
  258 |   Key firstPose;
      |       ^~~~~~~~~
```

## Expected behavior

<!-- A clear and concise description of what you expected to happen. -->
No warning.

## Environment

<!-- Please provide information about your code environment, things such as OS, language of use (C++, Matlab or Python), version of dependent libraries if using a custom build etc., anything to provide us more information. -->

<!-- The more the information, the faster we can help resolve the issue -->

OS: Arch Linux
GCC: 9.3.0
GTSAM: latest head on develop branch at time of posting:
b4eb0c2

## Additional information

<!-- Add any other infor or context about the problem here. -->

N/A

--- a/examples/SolverComparer.cpp
+++ b/examples/SolverComparer.cpp
@@ -255,7 +255,7 @@ void runIncremental()
   cout << "Looking for first measurement from step " << firstStep << endl;
   size_t nextMeasurement = 0;
   bool havePreviousPose = false;
-  Key firstPose;
+  Key firstPose = 0;
   while(nextMeasurement < datasetMeasurements.size())
   {
     if(BetweenFactor<Pose>::shared_ptr factor =

